### DISCUSSION TITLE
我为这个软件写了一个全自动压缩器，可以大大减少备份文件的大小

### DISCUSSION THREAD

**Discussion by ghost** [ShunCai/QZoneExport#56] [Ideas]
我为这个软件写了一个全自动压缩器，自动把图片全部压缩成BPG，BPG的 大小比JPEG小很多，然后自动修改HTML文件

**Comment by ShunCai** [comment]
备份还在意大小吗？共享来看看。

**Reply by ghost** [reply]
备份当然是越小越好啊，而且这是变小是不损失质量的，代码刚撸完有点累，休息一下就传到GITHUB和知乎

**Reply by ghost** [reply]
在非常保守的参数下，大小能变为原来的四分之一，以后如果加上自适应压缩还可以更小

**Reply by ghost** [reply]
也不是所有的都有四分之一的效果，有时候只有二分之一

**Reply by ShunCai** [reply]
heif格式更小，只是现在兼容性没那么好。

**Reply by ShunCai** [reply]
> 备份当然是越小越好啊，而且这是变小是不损失质量的，代码刚撸完有点累，休息一下就传到 GITHUB 和知乎

传了答复下，俺也用下。^_^

**Reply by ghost** [reply]
> heif格式更小，只是现在兼容性没那么好。

Bellard给的JS的BPG解码器太好用了，就用BPG了，对HTML的改动非常小，就是加上.bpg后缀再把date-src全部替换成src就可以用了。HEIC我一时找不到好用的解码器。

**Reply by ShunCai** [reply]
> > heif 格式更小，只是现在兼容性没那么好。
> 
> Bellard 给的 JS 的 BPG 解码器太好用了，就用 BPG 了，对 HTML 的改动非常小，就是加上.bpg 后缀再把 date-src 全部替换成 src 就可以用了。HEIC 我一时找不到好用的解码器。

为啥要替换data-src为src，我记得之前用data-src是为了实现图片懒加载，直接吧data-src的值换掉就ok了吧。

**Reply by ghost** [reply]
> > > heif 格式更小，只是现在兼容性没那么好。
> > 
> > 
> > Bellard 给的 JS 的 BPG 解码器太好用了，就用 BPG 了，对 HTML 的改动非常小，就是加上.bpg 后缀再把 date-src 全部替换成 src 就可以用了。HEIC 我一时找不到好用的解码器。
> 
> 为啥要替换data-src为src，我记得之前用data-src是为了实现图片懒加载，直接吧data-src的值换掉就ok了吧。

直接替换data-src的值不能正常显示，可能是库不支持的关系，我完全没有学过HTML，所以并不懂，修改也很简单，不需要HTML知识 也能改出来。

**Reply by ghost** [reply]
压缩器的下载地址，参数不懂就直接回车，目录填“C:\QQ空间备份2021.02.03\QQ空间备份\Messages”这种格式，显示图片需要把浏览器的禁止跨域访问关掉，不然无法访问本地的文件。CHROME我怎么也关不掉，我就用的FIREFOX，很快就关掉了。
链接：https://pan.baidu.com/s/1qlzO1iqECQgKwVbyhjDqDw 
提取码：love

**Reply by ghost** [reply]
现在只支持压缩“说说”，不支持压缩别的，因为我别的也不需要压缩，就暂时没有写这个功能。

**Reply by ghost** [reply]
我已经把软件传上来了，你收到了吗

**Reply by ShunCai** [reply]
收到了，刚回家忙，没在弄。

**Comment by LioDon** [comment]
说句实话，我有一个照片数量为661的相册，数据完整备份下来之后，文件夹的大小为2.5G。在文件夹里照片可以正常打开，但打开本地的HTML相应的相册页的时候，加载速度慢不说，照片还没显示全就崩溃了……打开一个相册基本要占用2G左右的内存。所以，可以在打开相册页的时候先预览缩略图，点击特定照片再打开原图么……？

**Reply by ghost** [reply]
> 说句实话，我有一个照片数量为661的相册，数据完整备份下来之后，文件夹的大小为2.5G。在文件夹里照片可以正常打开，但打开本地的HTML相应的相册页的时候，加载速度慢不说，照片还没显示全就崩溃了……打开一个相册基本要占用2G左右的内存。所以，可以在打开相册页的时候先预览缩略图，点击特定照片再打开原图么……？

相册备份功能我也没用过，但是我觉得相册备份不需要用HTML啊，直接用文件夹归类不好吗？你用的相册备份文件夹有归类吗？

**Reply by LioDon** [reply]
不用HTML就没法备份跟照片相关的评论信息，所以，我希望的最佳备份方式是按照空间显示的原样完全复制到本地。

**Reply by ghost** [reply]
> 不用HTML就没法备份跟照片相关的评论信息，所以，我希望的最佳备份方式是按照空间显示的原样完全复制到本地。

我相册不需要备份，所以我帮不了你。我用这个软件是备份日志和说说的。

**Reply by ShunCai** [reply]
> 说句实话，我有一个照片数量为661的相册，数据完整备份下来之后，文件夹的大小为2.5G。在文件夹里照片可以正常打开，但打开本地的HTML相应的相册页的时候，加载速度慢不说，照片还没显示全就崩溃了……打开一个相册基本要占用2G左右的内存。所以，可以在打开相册页的时候先预览缩略图，点击特定照片再打开原图么……？

我记得已经改成懒加载预览相片了啊，难道还会崩溃吗？
如果要单独加载预览图也行，还得再把功能调整下，腾讯预览图跟原图的话是分开存储的，如果需要单独加载预览图，还得把预览图给下载下来。

如果已经是懒加载还崩溃的话就得分页显示了，当初就是觉得分页显示麻烦，并且现在大部分人的电脑都上8G内存了，所以就没弄分页。

你是哪个版本备份的数据，是不是因为当时那个版本还没图片懒加载功能。

**Reply by ShunCai** [reply]
> > 不用HTML就没法备份跟照片相关的评论信息，所以，我希望的最佳备份方式是按照空间显示的原样完全复制到本地。
> 
> 我相册不需要备份，所以我帮不了你。我用这个软件是备份日志和说说的。

可以把你工具再改改，兼容所有类型就好，比如读取某个文件夹下的所有的html的img标签的图片地址，然后做转换就行。

**Reply by LioDon** [reply]
> > 说句实话，我有一个照片数量为661的相册，数据完整备份下来之后，文件夹的大小为2.5G。在文件夹里照片可以正常打开，但打开本地的HTML相应的相册页的时候，加载速度慢不说，照片还没显示全就崩溃了……打开一个相册基本要占用2G左右的内存。所以，可以在打开相册页的时候先预览缩略图，点击特定照片再打开原图么……？
> 
> 我记得已经改成懒加载预览相片了啊，难道还会崩溃吗？
> 如果要单独加载预览图也行，还得再把功能调整下，腾讯预览图跟原图的话是分开存储的，如果需要单独加载预览图，还得把预览图给下载下来。
> 
> 如果已经是懒加载还崩溃的话就得分页显示了，当初就是觉得分页显示麻烦，并且现在大部分人的电脑都上8G内存了，所以就没弄分页。
> 
> 你是哪个版本备份的数据，是不是因为当时那个版本还没图片懒加载功能。

当时那个1.0.8备份的，我笔记本内存8G。

**Reply by ghost** [reply]
内存不够可以开虚拟内存

**Comment by ghost** [comment]
这个全自动压缩器真是个馊主意，我准备做一个压缩文件，把里面的所有图片自动压缩成HEIC，解包的时候再自动转回原图片格式，这样比这个 压缩器通用多了

**Comment by k3-cat** [comment]
抱歉 打扰大家一下

我觉得考虑到 @ahava-666 做的事情  大家应该**谨慎**的讨论它贡献的代码
因为不明原因（说不明是因为我之前甚至都没有见过这一系列id）它跑我的其中一个repo里开了个issue喷了一顿  然后把标题和内容用无意义字符替换掉  又换了个用户名
这个行为简直是难以理解  很难相信这样的人有着开源社区所倡导的价值观

下面是截屏：
![image](https://user-images.githubusercontent.com/15878537/117548860-c0f0e700-b069-11eb-888c-34b164b69882.png)
![image](https://user-images.githubusercontent.com/15878537/117548864-c6e6c800-b069-11eb-835a-0ec00cd0af08.png)

**Reply by ghost** [reply]
我就是没有开源社区所倡导的价值观

**Reply by ghost** [reply]
而且我几个月就把所有社交帐号全换了，你做的行为对我一点影响都没有，而且看得懂我代码的人根本没有精力去看你这些垃圾问题

**Reply by ghost** [reply]
我信仰专制和独裁

**Reply by ghost** [reply]
不管我发了什么东西，我都希望大家谨慎讨论我发上来的代码，带脑判断是最基本的东西

**Reply by ghost** [reply]
多挂挂，多挂挂，省得我到处说你代码垃圾，你自己帮我宣传我都省力了

**Reply by k3-cat** [reply]
> 而且我几个月就把所有社交帐号全换了，你做的行为对我一点影响都没有，而且看得懂我代码的人根本没有精力去看你这些垃圾问题

看了下你的activity 没有pr 没有commit 连public repo 都没有 你的那些天才代码在哪
